src/lib/src/datepicker/datepicker.component.ts

```diff
@@ -22,16 +22,27 @@ const moment = (rawMoment as any).default ? (rawMoment as any).default : rawMome
 export class DatepickerComponent implements AfterViewInit {
   public static FORMAT = 'MM/DD/YYYY';
 
-  @Input() date: Date;
+  _date: Date;
+  @Input()
+  set date(value: Date) {
+    if (value && (
+      !this._date || (value && this._date.valueOf() !== value.valueOf()))) {
```

Comment:
this`value &&` now unnecessary

Reply:
I actually found that it is necessary because I got nonsense error messages in a metrics spec when value is undefined and value.valueOf() fails

Reply:
oh no, i mean now that you've added `value &&` on line 28, we don't need to have it on line 29 too. or do you mean you have found you need both?

Reply:
I'll rewrite it to be less confusing